### PR TITLE
feat: clear session if authentication methods change

### PR DESCRIPTION
**Pull request by yquansah**
This PR clears a session from local storage if a user happens to go from unauth'd Flipt to auth'd Flipt.

There is a little more nuance we need to consider here though, like if the user changes auth providers in their auth configuration. I think that case would warrant us to clear the session as well.

### CODE REVIEW

File: ui/src/components/SessionProvider.tsx [hunk 1]
```diff
@@ -27,8 +27,32 @@ export default function SessionProvider({
...
 
   useEffect(() => {
+    const clearSessionIfNecessary = async () => {
+      const config = await getConfig();
+      if (session && session.required === false) {
+        let authEnabled = false;
+        if (
+          config.authentication.methods.oidc.enabled ||
+          config.authentication.methods.github.enabled ||
+          config.authentication.methods.jwt.enabled
```
Comment (markphelps) [inline ui/src/components/SessionProvider.tsx:37]: im not sure if we need to check if they enabled a auth method without it being required right? i would think we only care if it goes from `auth.requred == false` to `auth.required == true` (and maybe the opposite as well?)

File: ui/src/components/SessionProvider.tsx [hunk 2]
```diff
@@ -27,8 +27,32 @@ export default function SessionProvider({
...
+          clearSession();
+        }
+      }
+    };
+
     const loadSession = async () => {
-      let session = {
+      if (session) {
+        clearSessionIfNecessary();
+        if (session) {
```
Comment (markphelps) [inline ui/src/components/SessionProvider.tsx:50]: is this block required? either way it would return right?

Comment (yquansah) [reply ui/src/components/SessionProvider.tsx:50]: @markphelps I didn't want the rest of the code in that function to run if there exists a session after we've tried to clear it

File: ui/src/components/SessionProvider.tsx (hunk 1 above)
Comment (yquansah) [reply ui/src/components/SessionProvider.tsx:37]: @markphelps Yeah I struggled with this. What does enabled but not required mean for the browser?

Comment (markphelps) [reply ui/src/components/SessionProvider.tsx:37]: nothing. if its not required then the API wont enforce it and the browser shouldn't care

File: ui/src/components/SessionProvider.tsx (hunk 2 above)
Comment (markphelps) [reply ui/src/components/SessionProvider.tsx:50]: ah yeah i didnt see that teh function continued

File: ui/src/components/SessionProvider.tsx [hunk 3]
```diff
@@ -27,8 +27,30 @@ export default function SessionProvider({
...
+        if (config.authentication.required) {
+          clearSession();
+        }
+      }
+
+      if (session && session.required === true) {
+        if (!config.authentication.required) {
+          clearSession();
+        }
+      }
```
Comment (markphelps) [inline ui/src/components/SessionProvider.tsx:42]:
```suggestion
      if (session && session.required === config.authentication.required) {
          clearSession();
      }
```

is ^ functionally the same?

**Review by markphelps — APPROVED** [review]
ty!

**Pull request merged**